### ISSUE TITLE
drawtriangles not working [native, neko, html5, Mac OSX]

### ISSUE THREAD

**Issue by Hasufel** [openfl/openfl#758]
Using source of today. Tested on MBP with Intel HD graphics running OS X.

Native:

- using Bunnymark sample in openfl-samples, the background - using drawtriangles - won't render. If we set the ```DrawTrianglesTest ()``` method on bunnymark instead of the ```TilesheetTest ()``` method, no bunnies will render.

Neko:

- using the same Bunnymark and ```DrawTrianglesTest ()``` method wil lead to a crash at runtime:

```Called from ? line 1
Called from ApplicationMain.hx line 123
Called from ApplicationMain.hx line 55
Called from lime/app/Application.hx line 163
Called from lime/_backend/native/NativeApplication.hx line 107
Called from a C function
Called from lime/app/Event.hx line 58
Called from lime/app/Application.hx line 568
Called from openfl/display/Stage.hx line 986
Called from openfl/_internal/renderer/opengl/GLRenderer.hx line 299
Called from openfl/_internal/renderer/opengl/GLRenderer.hx line 330
Called from openfl/display/DisplayObjectContainer.hx line 977
Called from openfl/display/DisplayObjectContainer.hx line 977
Called from openfl/display/DisplayObjectContainer.hx line 977
Called from openfl/display/DisplayObjectContainer.hx line 973
Called from a C function
Called from openfl/display/DisplayObject.hx line 1213
Called from openfl/_internal/renderer/opengl/utils/GraphicsRenderer.hx line 820
Called from openfl/_internal/renderer/opengl/utils/GraphicsRenderer.hx line 1000
Called from openfl/_internal/renderer/opengl/utils/GraphicsRenderer.hx line 692
Uncaught exception - Invalid array access```

HTML5: 
- using the same Bunnymark and ```DrawTrianglesTest ()``` method, no bunnies will render.

Flash: renders perfectly fine.

**Comment by larsiusprime** [comment]
My system:
```
  OS : Windows 7 SP1
  RAM: 8387064 KB (8 GB)
  CPU: Intel(R) Core(TM)2 Duo CPU     E7400  @ 2.80GHz
  GPU: ATI Radeon HD 4800 Series, driver v. 8.970.100.1100
```

Results:

BLANK = expected results
NBY = No bunnies
CRASH = Crash

(From the looks of it the background code has been disabled in the latest demo code? So I'm not sure "no background" -- which was true for all the demos -- is a significant finding...)

|               | Tilesheet | DrawRect | Bitmap | DrawTriangles | CopyPixels | NoBatchTilesheet |
|-------------|------------|--------------|----------|--------------------|----------------|------------------------|
| Windows |               |                  |            |        NBY          |                    |                              |
| Neko       |               |                  |            |      CRASH       |                    |                              |
| Flash       |               |                  |            |                         |                    |                              |
| HTML5    |               |                  |            |        NBY          |                    |                              |

Windows and HTML5 have no bunnies for DrawTriangles, Neko crashes on DrawTriangles,  Flash is fine across everything, all other tests seem fine.

**Comment by Hasufel** [comment]
The background is supposed to always be using drawtiles by default.

**Comment by ibilon** [comment]
I looked into this and I don't get it at all.
After downloading an old version of BunnyMark and uncommenting the background code I got it working as in flash in native (linux), nothing on screen for html5 and crashing for neko.

Looked into the neko crash, it's this line https://github.com/openfl/openfl/blob/develop/openfl/_internal/renderer/cairo/CairoGraphics.hx#L797
but I traced `iax` and it's 0, `v` does indeed contain the data, but still invalid array access,
even tried using `Array` inside `Vector` instead of `haxe.ds.Vector` but it also crash in the access.

So weird.
